etl-pipeline/managers/sqs.py

```diff
@@ -0,0 +1,147 @@
+import json
+import boto3
+from botocore.exceptions import ClientError
+from typing import Union
+import os
+from pika.exceptions import StreamLostError
+
+from managers import RabbitMQManager
+from logger import create_log
+
+logger = create_log(__name__)
+
+
+class SQSManager:
+    def __init__(self, visibility_timeout=30, max_receive_count=1):
+        super(SQSManager, self).__init__()
+        self.region_name = os.environ.get("AWS_REGION", "us-east-1")
+        self.endpoint_url = os.environ.get("S3_ENDPOINT_URL")
+        self.aws_access_key_id = os.environ.get("AWS_ACCESS_KEY_ID")
+        self.aws_secret_access_key = os.environ.get("AWS_SECRET_ACCESS_KEY")
+        self.queue_name = os.environ.get("RECORD_PIPELINE_SQS_QUEUE", "records")
+        self.visibility_timeout = visibility_timeout
+        self.max_receive_count = max_receive_count
+        self.client = self.create_client()
+        self.queue_url = None
+        self.rabbit_route = os.environ.get("RECORD_PIPELINE_ROUTING_KEY", 'records')
+        self.rabbit_mq = RabbitMQManager()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_tb):
+        pass  # boto3 client doesn't need explicit cleanup
+
+    def create_client(self):
+        client_params = {
+            "region_name": self.region_name,
+        }
+
+        if self.endpoint_url:
+            client_params["endpoint_url"] = self.endpoint_url
+
+        if self.aws_access_key_id and self.aws_secret_access_key:
+            client_params["aws_access_key_id"] = self.aws_access_key_id
+            client_params["aws_secret_access_key"] = self.aws_secret_access_key
+
+        self.client = boto3.client("sqs", **client_params)
+
+    def get_queue(self) -> boto3.resources.base.ServiceResource:  # type: ignore
+        """Get the SQS queue resource with type annotations"""
+        if not self.client:
+            self.create_client()
+            
+        assert self.client is not None  # Tell type checker client is initialized
+        
+        try:
+            response = self.client.get_queue_by_queue_name(QueueName=self.queue_name)
+            self.queue_url = response["QueueUrl"]
+            return response
+        except ClientError as e:
+            logger.error(f"Failed to connect to SQS queue: {e}")
+            raise
+
+    def send_message_to_queue(self, message: Union[str, dict]):
+        queue = self.get_queue()
+        if isinstance(message, dict):
+            message = json.dumps(message)
+
+        params = {"MessageAttributes": message}
+
+        try:
+            response = queue.send_message(**params)
```

Comment:
We could also pass in the queue url here rather than making a call to get the queue.